### ISSUE TITLE
Issue with breadcrumb

### ISSUE THREAD

**Issue by sthada** [OfficeDev/office-ui-fabric-react#9557]
Issue faced:

My state variable is:

```ts
items: [
        { text: 'Home', key: 'f0', onClick: this._onBreadcrumbItemClicked, isCurrentItem: true },
        { text: 'Create Project', key: 'f1', onClick: this._onBreadcrumbItemClicked, isCurrentItem: false },
        { text: 'Select Tables & Generate Code', key: 'f2', onClick: this._onBreadcrumbItemClicked, isCurrentItem: false },
        { text: 'Download project Repositort', key: 'f3', onClick: this._onBreadcrumbItemClicked, isCurrentItem: true } 
      ];
```

```tsx
<Breadcrumb className='bre-pos'
        items={this.state.items}
        maxDisplayedItems={4}
        onReduceData={this.navigateThrough}
        ariaLabel={'Breadcrumb with static width'}
        overflowAriaLabel={'More items'}
/>
```

I want to remove all the elements coming after the clicked one (e.g. if I click on breadcrumb with key 'f1', the breadcrumb to be displayed must be with keys 'f0' and 'f1' and the breadcrumb NOT to be displayed are with keys 'f2' and 'f3' ).

Apart from it i am not able to set State in `_onBreadcrumbItemClicked` event as i want to navigate and render on the basis of breadcrumb clicked.

Can you help me out with this?

**Comment by KevinTCoughlin** [comment]
Before I explore your ask I'd like to make sure I fully understand it. My understanding is:

You would like an example showing how to hide `Breadcrumb` options after an option is selected. In this case hiding / removing the options which have a key that comes after the option selected.

Is the above correct?

Have you tried recreating the `items` array after a user action?

**Comment by sthada** [comment]
yes @Kevin i tried that and the above is correct. I am not able to call function or use setState inside **onReduceData={this.navigateThrough}** or **_onBreadcrumbItemClicked** .

Apart from this i can not redirect the page or use routing inside any method.
i am trying with this.props.history.push('/page1');

**Comment by sthada** [comment]
Hi @KevinTCoughlin , let me rephrase what i am trying to do, i want to use routing or navigate to different page using breadcrumb, Can you suggest me how to do that.

**Comment by mirajp** [comment]
@sthada when you say you aren't able to setState inside the `_onBreadcrumbItemClicked` function, do you mean you get an undefined error for `this.setState`? If you're getting an undefined reference error, you may need to bind the function in your parent Component/page class's constructor so that on click `this` scope is that of the parent React Component (or simply use an arrow function); refer to:
https://reactjs.org/docs/faq-functions.html
https://medium.com/silesis/handle-events-in-react-with-arrow-functions-ede88184bbb
https://medium.com/@nikolalsvk/loosing-bind-this-in-react-8637ebf372cf

You can then use the `history` package to listen to route changes (once you create a shareable/importable and single instance of BrowserHistory) and do something like
`setState({ items: getCrumbsFromPath(window.location.pathname) });` whenever the route changes: https://github.com/ReactTraining/history#usage
Note that parsing route changes for routes with route parameters (e.g. '/user/:id') is a little more complicated but you should be able to find numerous resources/approaches online.
